### PR TITLE
Feature/user delete own account

### LINKED ISSUE #301: Allow users to delete his own account
Requested by Apple

### PR DESCRIPTION
**Pull request by tomivm**
close #301

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -247,7 +249,19 @@ async function listUser(req, res) {
...
 function removeUser(req, res) {
   const id = req.swagger.params.id.value;
-  User.findByIdAndRemove(id, function (err, users) {
+
+  //this would be implemented like a middleware
+  const { requestedBy, isAdmin: isRequestedByAdmin } = getAuthDataFromReq(req);
+
+  if (!isRequestedByAdmin && (!requestedBy || id != requestedBy)) {
+    return res.status(401).json({
+      message: 'Error getting subscriber',
```
Comment (RodriSanchez1) [inline api/controllers/user.js:258]: Change the error message

File: api/controllers/account.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+async function removeAccount(req, res) {
+  const id = req.swagger.params.id.value;
+
+  //this would be implemented like a middleware
+  const { requestedBy, isAdmin: isRequestedByAdmin } = getAuthDataFromReq(req);
+
+  if (!isRequestedByAdmin && (!requestedBy || id != requestedBy)) {
+    return res.status(401).json({
+      message: 'Error getting subscriber',
```
Comment (RodriSanchez1) [inline api/controllers/account.js:21]: Change the error message

File: api/controllers/user.js [hunk 3]
```diff
@@ -247,7 +249,19 @@ async function listUser(req, res) {
 
 function removeUser(req, res) {
```
Comment (RodriSanchez1) [inline api/controllers/user.js:250]: You are not using this function on the feature.  Isnt it?

Comment (tomivm) [reply api/controllers/user.js:250]: Yes. These changes are unnecessary now.

**Pull request merged**